source/grid/tria.cc

```diff
@@ -10494,8 +10500,23 @@ Triangulation<dim, spacedim>::create_triangulation(
   // because sometimes other objects are already attached to it:
   try
     {
-      internal::TriangulationImplementation::Implementation::
-        create_triangulation(v, cells, subcelldata, *this);
+      const bool arbitray_mesh_provided =
+        std::any_of(cells.begin(), cells.end(), [](const auto &cell) {
+          return cell.vertices.size() != GeometryInfo<dim>::vertices_per_cell;
+        });
+
+      if (arbitray_mesh_provided == false)
+        {
+          internal::TriangulationImplementation::Implementation::
+            create_triangulation(v, cells, subcelldata, *this);
+        }
+      else
+        {
+          Assert(
+            false,
```

Comment:
I think it would be easier to read if you did
```
  AssertThrow (std::any_of (...) == false, ExcMessage(...));
```
This way you don't need to create the variable.

It should be `AssertThrow` so that things read from input files are also captured.

Reply:
I would like to keep it this way. In a follow-up PR, I might save the bool locally and in the else statement I might replace the `Assert(Throw)`.